public/js/dashboardCategories.js

```diff
@@ -68,6 +68,69 @@ window.deleteCategory = function deleteCategory(categoryID) {
   console.log('deleted' + categoryID);
 };
 
+/**
+ * Adds an input box to the correct section of the page.
+ * @param {string} prevInstructionID ID of the previous input
+ * box's add button [instructionType]-add-[instructionIndex].
+ */
+function addInstructionInput(prevInstructionID) {
+  const prevInstructionIdArr = prevInstructionID.split('-');
+  const instructionType = prevInstructionIdArr[0];
+  const currInstructionIndex = parseInt(prevInstructionIdArr[2]) + 1;
+
+  const prevRemoveButton = document.getElementById(
+    instructionType + '-remove-' + prevInstructionIdArr[2]
+  );
+  if (prevRemoveButton) prevRemoveButton.hidden = false;
+
+  const prevAddButton = document.getElementById(
+    instructionType + '-add-' + prevInstructionIdArr[2]
+  );
+  if (prevAddButton) prevAddButton.hidden = true;
+
+  const currInstructionDiv = document.createElement('div');
+  currInstructionDiv.setAttribute('id', instructionType + '-container-' + currInstructionIndex);
+
+  const inputBox = document.createElement('input');
+  inputBox.setAttribute('id', instructionType + '-input-' + currInstructionIndex);
+  inputBox.setAttribute('type', 'text');
+  inputBox.setAttribute('name', instructionType + '-input-' + currInstructionIndex);
+
+  const removeButton = document.createElement('button');
+  removeButton.innerHTML = '-';
+  removeButton.setAttribute('id', instructionType + '-remove-' + currInstructionIndex);
+  removeButton.setAttribute('onclick', 'removeInstructionInput(this.id)');
+  removeButton.hidden = true;
+
+  const addButton = document.createElement('button');
+  addButton.innerHTML = '+';
+  addButton.setAttribute('id', instructionType + '-add-' + currInstructionIndex);
+  addButton.setAttribute('onclick', 'addInstructionInput(this.id)');
+
+  const breakElement = document.createElement('br');
+
+  currInstructionDiv.append(inputBox);
+  currInstructionDiv.append(removeButton);
+  currInstructionDiv.append(addButton);
+  currInstructionDiv.append(breakElement);
```

Comment:
nit: An idea here is that you can make a handlebars template to construct the elements of instructionContainer rather than with JS. Similiar to how Sabrina did it with the search cards.

see `public/js/searchList` [here ](url)

Reply:
You could also try making a function that creates an element and sets its attributes/parents/innerText/etc.

Reply:
Yea, I agree that doing Malcolm's suggestion or making a function to handle creating elements would make this section less repetitive.